src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2901,19 +2902,40 @@ private BlockType findKnownBlock(void* location, out int offset)
 
         partial void findKnownBBCountBlock(ref BlockType blockType, void* location, ref int offset);
 
+        private static RelocType GetRelocType(TargetArchitecture targetArchitecture, ushort fRelocType)
+        {
+            if (targetArchitecture != TargetArchitecture.ARM64)
+                return (RelocType)fRelocType;
+
```

Comment:
This function needs a comment describing what it is doing. (Why the relocs for arm64 are not mapped to the same numeric values in the JIT as opposed to the crossgen2 codebase.)

Reply:
I added the comment: "Translates relocation type constants used by JIT (defined in winnt.h) to our RelocType enumeration".  However, there is no need to use different constants for these two relocation types.  We can change their values in RelocType enumeration to be 4 and 6 as in JIT.  Then we don't need to map them.  I am not sure why we didn't use 4 and 6 in RelocType from the beginning.

With these changes compilation for ARM64 finishes successfully and generates a R2R image.  There are still issues regarding execution, which need addressing.